src/NLog/Targets/MailTarget.cs

```diff
@@ -408,10 +423,10 @@ private void ConfigureMailClient(LogEventInfo lastEvent, ISmtpClient client)
 
         private void CheckRequiredParameters()
         {
-            if (!this.UseSystemNetMailSettings && this.SmtpServer == null)
+            if (!this.UseSystemNetMailSettings && this.SmtpServer == null && string.IsNullOrEmpty(this.PickupDirectoryLocation))
```

Comment:
That's right, If ` UseSystemNetMailSettings` is true we will expect the `mailSettings` to be in the application `config` file. However if that is not the case we can not have both the `Host` and `PickupDirectoryLocation` as null.